crates/common/src/state_update.rs

```diff
@@ -348,25 +353,107 @@ mod state_diff_commitment {
         system_contract_updates: &HashMap<ContractAddress, SystemContractUpdate>,
         declared_cairo_classes: &HashSet<ClassHash>,
         declared_sierra_classes: &HashMap<SierraHash, CasmHash>,
+        version: StarknetVersion,
     ) -> StateDiffCommitment {
-        StateDiffCommitment(
-            poseidon_hash_many(&[
-                // state_diff_version
-                MontFelt::ZERO,
-                compute_hash_of_deployed_contracts(contract_updates),
-                compute_hash_of_declared_classes(declared_sierra_classes),
-                compute_hash_of_old_declared_classes(declared_cairo_classes),
-                // number_of_DA_modes
-                MontFelt::ONE,
-                // DA_mode_0
-                MontFelt::ZERO,
-                compute_hash_of_storage_domain_state_diff(
-                    contract_updates,
-                    system_contract_updates,
-                ),
-            ])
-            .into(),
-        )
+        if version < StarknetVersion::new(0, 13, 2, 0) {
+            StateDiffCommitment(
+                poseidon_hash_many(&[
+                    // state_diff_version
+                    MontFelt::ZERO,
+                    compute_hash_of_deployed_contracts(contract_updates),
+                    compute_hash_of_declared_classes(declared_sierra_classes),
+                    compute_hash_of_old_declared_classes(declared_cairo_classes),
+                    // number_of_DA_modes
+                    MontFelt::ONE,
+                    // DA_mode_0
+                    MontFelt::ZERO,
+                    compute_hash_of_storage_domain_state_diff(
+                        contract_updates,
+                        system_contract_updates,
+                    ),
+                ])
+                .into(),
+            )
+        } else {
+            static STARKNET_STATE_DIFF0: OnceLock<MontFelt> = OnceLock::new();
+            let starknet_state_diff0 = STARKNET_STATE_DIFF0
+                .get_or_init(|| MontFelt::from_hex(&hex::encode("STARKNET_STATE_DIFF0")));
```

Comment:
Can't we just use `const STARKNET_STATE_DIFF0: Felt = felt_bytes!(b"STARKNET_STATE_DIFF0"));` and do the Montfelt conversion in `hasher.write()` below? We're doing that conversion anyway for all other felts except for this first one so I don't think the performance impact would be substantial -- yet it would allow us to get rid of both `hex` and `OnceLock`.